src/AutoRest.CSharp/Common/Generation/Writers/ModelWriter.cs

```diff
@@ -84,198 +84,145 @@ protected virtual void WriteProperties(CodeWriter writer, ObjectType schema)
         {
             foreach (var property in schema.Properties)
             {
-                Stack<ObjectTypeProperty> hierarchyStack = property.GetHeirarchyStack();
-                if (Configuration.AzureArm && hierarchyStack.Count > 1)
+                WriteProperty(writer, property);
+
+                if (property.FlattenedProperty != null)
+                    WriteProperty(writer, property.FlattenedProperty);
+            }
+        }
+
+        private static bool AllCtorsContainMyProperty(ObjectType enclosingType, ObjectTypeProperty property)
```

Comment:
Do we need this method?

Reply:
I just moved these methods around in the same file. If no one is using it, I will remove it